sdk/storage/Azure.Storage.Blobs/tests/BlobBaseClientTests.cs

```diff
@@ -1410,6 +1411,70 @@ public async Task OpenReadAsync_Seek_NewPositionGreaterThanBlobLength_AllowBlobM
             Assert.AreEqual(size + 10, outputStream.Position);
         }
 
+        [Test]
+        public async Task OpenReadAsync_Seek_NewPositionGreaterThanBlobLength_AllowBlobModificationsTrue_NoData()
+        {
+            int size = Constants.KB;
+            await using DisposingContainer test = await GetTestContainerAsync();
+
+            // Arrange
+            var data = GetRandomBuffer(size);
+            BlockBlobClient blob = InstrumentClient(test.Container.GetBlockBlobClient(GetNewBlobName()));
+            using Stream stream = new MemoryStream(data);
+            await blob.UploadAsync(stream);
+
+            BlobOpenReadOptions options = new BlobOpenReadOptions(allowModifications: true);
+
+            // Act
+            Stream outputStream = await blob.OpenReadAsync(options: options).ConfigureAwait(false);
+            outputStream.Seek(size + 10, SeekOrigin.Begin);
+            byte[] outputArray = new byte[1];
+
+            // Assert
+            Assert.Zero(stream.Read(outputArray, 0, 1));
+        }
+
+        [Test]
+        public async Task OpenReadAsync_Seek_NewPositionGreaterThanBlobLength_AllowBlobModificationsTrue_Data()
+        {
+            int size = Constants.KB;
+            await using DisposingContainer test = await GetTestContainerAsync();
+
+            // Arrange
+            var data = GetRandomBuffer(size);
+            AppendBlobClient blob = InstrumentClient(test.Container.GetAppendBlobClient(GetNewBlobName()));
+            await blob.CreateAsync();
+            using Stream stream = new MemoryStream(data);
+            await blob.AppendBlockAsync(stream);
+
+            BlobOpenReadOptions options = new BlobOpenReadOptions(allowModifications: true);
+
+            // Act
+            Stream outputStream = await blob.OpenReadAsync(options: options).ConfigureAwait(false);
```

Comment:
If you don't read anything, does this change the size before you seek?

Reply:
Yes, we do a Get Properties request to make sure the blob exists and get it's initial length.